### ISSUE TITLE
Stack overflow, high CPU usage after upgrade to 0.17.1

### ISSUE THREAD

**Issue by SethDavenport** [DataDog/dd-trace-js#870]
**Describe the bug**

I recently updated dd-trace-js to 0.17.1 from 0.16.3 in order to consume this fix: https://github.com/DataDog/dd-trace-js/pull/844

I can confirm that 0.17.1 fixes the issue with koa-router, but once I deployed to production I immediately noticed much higher CPU usage in our web containers:

![image](https://user-images.githubusercontent.com/4614141/75004279-97ce4180-5438-11ea-8eeb-d33571a98a79.png)

The uptick at 6:25pm in that graph is the deployment that took us from 0.16.3 to 0.17.1; the return to normal CPU usage at around 6:40pm coincides with me rolling back to 0.16.3 again.

During this period of time a bunch of these came through in our error reporting system:

![image](https://user-images.githubusercontent.com/4614141/75004403-fe535f80-5438-11ea-8ec3-b5d2e20c8e8d.png)

A stack overflow would definitely chew through CPU so it seems related, not just because of the timing.

**Environment**

* **Operating system:** debian:buster-slim (docker container)
* **Node version:** 12.15.0
* **Tracer version:** 0.17.1

**Comment by rochdev** [comment]
What version of Koa are you using?

**Comment by SethDavenport** [comment]
![image](https://user-images.githubusercontent.com/4614141/75060078-2f20ac80-54ac-11ea-8487-df759fcf1d7a.png)

(2.11.0)

**Comment by SethDavenport** [comment]
Also not sure if relevant but here's the rest of the koa-related stuff:

```
koa-helmet@5.2.0
koa-body@4.1.1
koa-convert@1.2.0
koa-router@8.0.8
koa-websocket@6.0.0
```

**Comment by ffandreassoroko** [comment]
same problem

**Comment by rochdev** [comment]
I'm pretty sure the regression was introduced in #844 but I can't seem to reproduce. I'll try to do some testing with the code directly, but in the meantime is this something you can reproduce easily with a small snippet or only in a real app?

**Comment by rochdev** [comment]
Never mind, the problem is we make only 1 request per unit test, and this only happens with multiple requests on the same router. Working on a fix now.

**Comment by SethDavenport** [comment]
Sorry got sidetracked - glad you figured it out!

**Comment by rochdev** [comment]
Fix released in 0.18.1.

**Comment by SethDavenport** [comment]
Thanks for the quick turnaround.
